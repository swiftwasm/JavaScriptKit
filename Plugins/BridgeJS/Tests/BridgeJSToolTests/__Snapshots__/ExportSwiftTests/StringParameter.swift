// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(BridgeJS) import JavaScriptKit

@_expose(wasm, "bjs_checkString")
@_cdecl("bjs_checkString")
public func _bjs_checkString(aBytes: Int32, aLength: Int32) -> Void {
    #if arch(wasm32)
    checkString(a: String.bridgeJSLiftParameter(aBytes, aLength))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_roundtripString")
@_cdecl("bjs_roundtripString")
public func _bjs_roundtripString(aBytes: Int32, aLength: Int32) -> Void {
    #if arch(wasm32)
    let ret = roundtripString(a: String.bridgeJSLiftParameter(aBytes, aLength))
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}