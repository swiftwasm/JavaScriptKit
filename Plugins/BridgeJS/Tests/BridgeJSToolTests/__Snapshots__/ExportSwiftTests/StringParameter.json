{
  "classes" : [

  ],
  "enums" : [

  ],
  "functions" : [
    {
      "abiName" : "bjs_checkString",
      "effects" : {
        "isAsync" : false,
        "isStatic" : false,
        "isThrows" : false
      },
      "name" : "checkString",
      "parameters" : [
        {
          "label" : "a",
          "name" : "a",
          "type" : {
            "string" : {

            }
          }
        }
      ],
      "returnType" : {
        "void" : {

        }
      }
    },
    {
      "abiName" : "bjs_roundtripString",
      "effects" : {
        "isAsync" : false,
        "isStatic" : false,
        "isThrows" : false
      },
      "name" : "roundtripString",
      "parameters" : [
        {
          "label" : "a",
          "name" : "a",
          "type" : {
            "string" : {

            }
          }
        }
      ],
      "returnType" : {
        "string" : {

        }
      }
    }
  ],
  "moduleName" : "TestModule",
  "protocols" : [

  ]
}